localstack/services/lambda_/invocation/event_manager.py

```diff
@@ -299,6 +270,36 @@ def handle_message(self, message: dict) -> None:
                 "Error handling lambda invoke %s", e, exc_info=LOG.isEnabledFor(logging.DEBUG)
             )
 
+    def process_event_retry(self, sqs_invocation: SQSInvocation, error: Exception):
```

Comment:
nit: The `event_retry` suffix could be a bit misleading because the extracted code only covers "Throttles (429) and system errors (5xx)". `process_throttles_and_system_errors` would be more accurate here, but longer.

"Function errors (returned from code or runtime, such as timeouts)" are the real "event retries" but are handled in the code below. Retry handling differs depending on the error: https://aws.amazon.com/blogs/compute/introducing-new-asynchronous-invocation-metrics-for-aws-lambda/